lib/connections/service.go

```diff
@@ -437,18 +437,19 @@ func (s *Service) VerifyConfiguration(from, to config.Configuration) error {
 }
 
 func (s *Service) CommitConfiguration(from, to config.Configuration) bool {
-	// We require a restart if a device as been removed.
-
-	restart := false
-
 	newDevices := make(map[protocol.DeviceID]bool, len(to.Devices))
 	for _, dev := range to.Devices {
 		newDevices[dev.DeviceID] = true
 	}
 
 	for _, dev := range from.Devices {
 		if !newDevices[dev.DeviceID] {
-			restart = true
+			s.curConMut.Lock()
+			delete(s.currentConnection, dev.DeviceID)
```

Comment:
So from glancing at it due to the other pull request, I think it actually doesn't really work. Since we don't know when a connection closes we'll never accept a relay connection after having had a direct connection for example. Also we'll log the "switching connections" message and the "Connection to ... closed" messages even when there is no current connection, as long as we've had a relay connection at some point